### PR TITLE
fix quickpulse url construction.

### LINKED ISSUE #1089: QuickPulse regression in 2.5.1-beta
The QuickPulse urls are malformed. From the http logs:
```http
POST /QuickPulseService.svcping?ikey=00000000-0000-0000-0000-000000000000 HTTP/1.1
```
(missing slash before 'ping')

### PR DESCRIPTION
**Pull request by littleaj**
add tests to verify.

Fix #1089

### CODE REVIEW

**Comment by littleaj** [comment]
@trask, I had to make quite a few changes to prevent static variable values from bleeding into other tests which were causing failures. Ping me if you want me to go through the details.

File: core/src/test/java/com/microsoft/applicationinsights/extensibility/initializer/docker/DockerContextInitializerTests.java [hunk 1]
```diff
@@ -38,10 +39,8 @@
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
-import org.junit.Assert;
-import org.junit.Before;
-import org.junit.BeforeClass;
-import org.junit.Test;
+
+import org.junit.*;
```
Comment (trask) [inline core/src/test/java/com/microsoft/applicationinsights/extensibility/initializer/docker/DockerContextInitializerTests.java]: Did you change Intellij settings? I think default is 99 imports before it uses '*'

File: core/src/test/java/com/microsoft/applicationinsights/internal/heartbeat/HeartbeatTests.java [hunk 2]
```diff
@@ -16,19 +17,22 @@
 import java.util.concurrent.Callable;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ConcurrentMap;
-import org.junit.AfterClass;
-import org.junit.Assert;
-import org.junit.Before;
-import org.junit.Test;
+
+import org.junit.*;
```
Comment (trask) [inline core/src/test/java/com/microsoft/applicationinsights/internal/heartbeat/HeartbeatTests.java]: Same here re: Intellij settings

File: core/src/test/java/com/microsoft/applicationinsights/internal/quickpulse/DefaultQuickPulseDataFetcherTests.java [hunk 3]
```diff
@@ -1,11 +1,15 @@
 package com.microsoft.applicationinsights.internal.quickpulse;
 
 import com.microsoft.applicationinsights.TelemetryConfiguration;
-import org.junit.Assert;
-import org.junit.Test;
+import org.junit.*;
```
Comment (trask) [inline core/src/test/java/com/microsoft/applicationinsights/internal/quickpulse/DefaultQuickPulseDataFetcherTests.java]: Same here re: Intellij settings

File: core/src/test/java/com/microsoft/applicationinsights/extensibility/initializer/docker/DockerContextInitializerTests.java (hunk 1 above)
Comment (littleaj) [reply core/src/test/java/com/microsoft/applicationinsights/extensibility/initializer/docker/DockerContextInitializerTests.java]: Yes. For tests, I allow * for `org.junit`, `org.hamcrest` and `org.mockito` and there are a few static ones too, e.g. `org.junit.Assert.*`. For prod code, I prefer to be explicit. For tests I like the splats for the test framework classes. Intellisense is usually quicker/better with its suggestions.

I think the checkstyle settings reflect this, but maybe not.

Comment (trask) [reply core/src/test/java/com/microsoft/applicationinsights/extensibility/initializer/docker/DockerContextInitializerTests.java]: How do you configure this differently in Intellij for test vs prod?

Comment (littleaj) [reply core/src/test/java/com/microsoft/applicationinsights/extensibility/initializer/docker/DockerContextInitializerTests.java]: I don't think you can. I have the count settings set to 500 (default is 99 but that didn't always work for me), plus rules for org.junit, Mockito, Matchers and Assert which always use `*`. These settings are under Editor/Code Style/Java/Imports

**Pull request merged**